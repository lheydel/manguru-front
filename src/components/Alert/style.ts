import { red } from '@material-ui/core/colors';
import makeStyles from '@material-ui/styles/makeStyles';

const commonStyle = {
    padding: '10px 15px',
    borderRadius: 3,
};

export default makeStyles({
    danger: {
        ...commonStyle,
        color: red.A700,
        background: red[50],
        borderColor: red[100],
    }
});
