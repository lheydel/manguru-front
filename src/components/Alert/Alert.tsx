import { Box } from '@material-ui/core';
import { BoxProps } from '@material-ui/core/Box';
import React from 'react';
import style from './style';

export interface AlertProps {
  variant: 'danger';
}

/**
 * Display a message within a colored block
 */
export const Alert: React.FC<AlertProps & BoxProps> = ({ children, variant, ...props }) => {
  const classes = style();

  return (
    <Box
      className={classes[variant]}
      border={1}
      {...props}
    >
      {children}
    </Box>
  );
};
