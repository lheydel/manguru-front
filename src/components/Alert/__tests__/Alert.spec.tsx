import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Alert from '..';
import { AlertProps } from '../Alert';

describe('Alert component', () => {
  const child = <i>blblbl</i>;

  let wrapper: ReturnType<typeof mount>;
  const mountComp = (variant: AlertProps['variant'] = 'danger') => {
    return mount(<Alert variant={variant}>{child}</Alert>);
  };

  afterEach(() => {
    wrapper.unmount();
  });

  test.each`
    variant
    ${'danger'}
  `('variant: [$variant]', (variant) => {
    wrapper = mountComp(variant);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render its children', () => {
    wrapper = mountComp();
    expect(wrapper.find(child)).toBeDefined();
  });
});
