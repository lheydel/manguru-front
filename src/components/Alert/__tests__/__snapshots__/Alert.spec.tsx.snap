// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component variant: [danger] 1`] = `
<Alert
  variant={
    Object {
      "variant": "danger",
    }
  }
>
  <Styled(MuiBox)
    border={1}
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <i>
        blblbl
      </i>
    </div>
  </Styled(MuiBox)>
</Alert>
`;
